docs/developer_manual/services/advanced/service_updater_upgrade.md

```diff
@@ -0,0 +1,269 @@
+
+We invite you to look at the `ServiceUpdater` code to have a better understanding of how the new service updaters work and how they use the functions you implement (`is_valid`, `import_update`).
+
+## Do I have to change the service manifest for 4.1?
+Yes, very much so. We've gone away from considering the service updater as a special dependency that needs it's own config section.
+
+We now consider it just be part of the service's list of dependencies, although we reserve the container name 'updates' to indicate this dependency is an updater to the service.
+
+
+Service Manifest for 4.0:
+```yaml
+
+update_config:
+  generates_signatures: true
+  method: run
+  run_options:
+    allow_internet_access: true
+    command: ["python", "-m", "sigma_updater"]
+    image: ${REGISTRY}cccs/assemblyline-service-sigma:$SERVICE_TAG
+  sources:
+    - name: sigma
+      pattern: .*windows\/.*\.yml
+      uri: https://github.com/SigmaHQ/sigma.git
+  update_interval_seconds: 21600 # Quarter-day (every 6 hours)
+```
+
+will now become in 4.1:
+```yaml
+dependencies:
+  updates:
+    container:
+      allow_internet_access: true
+      command: ["python", "-m", "sigma_.update_server"]
+      image: ${REGISTRY}cccs/assemblyline-service-sigma:$SERVICE_TAG
+      ports: ["5003"]
+    run_as_core: True
+
+update_config:
+  generates_signatures: true
+  sources:
+    - name: sigma
+      pattern: .*windows\/.*\.yml
+      uri: https://github.com/SigmaHQ/sigma.git
+  update_interval_seconds: 21600 # Quarter-day (every 6 hours)
+  signature_delimiter: "file"
+```
+
+Notes:
+ - Source management will still be part of the update_config, but container configuration will be moved to a list of dependencies.
+ - `signature_delimiter` indicates how signatures should be separated when downloaded from Assemblyline's Signature API (default: double newline).
+    - From Sigma's perspective, it gets it's a list of files of the form: `/updates/<random_tempdir>/sigma/<source_name>/<signature_name>` where signature_name only contains the signature associated to the name
+    - Under the default delimiter, it would've gotten: `/updates/<random_tempdir>/sigma/<source_name>` where source_name was a compiled list of all signatures from the source separated by double newlines in a single file (which might be fine for some service like YARA and Suricata)
+
+ - In order for updaters to work, they need to be able to communicate with other core components. So you'll need to enable the dependency to be able to `run_as_core`, otherwise this could lead to issues where the updater isn't able to resolve to components like Redis and/or Elasticsearch by name.
+ - Setting port(s) helps facilitate communication between the service and it's dependency over certain ports and so, as a result, Scaler will create the appropriate NetworkPolicy to ensure communication **only** between a service and its dependencies.
```

Comment:
```suggestion
 - Setting port(s) helps facilitate communication between the service and its dependency. As a result, Scaler will create the appropriate NetworkPolicy to ensure communication **only** between a service and its dependencies.
```

"to ensure communication only between" seems wrong to me, but I don't know what it should be.